modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/order/status/QuoteProcessedCommerceOrderStatusImpl.java

```diff
@@ -0,0 +1,155 @@
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.internal.order.status;
+
+import com.liferay.commerce.configuration.CommerceOrderFieldsConfiguration;
+import com.liferay.commerce.constants.CommerceConstants;
+import com.liferay.commerce.constants.CommerceOrderConstants;
+import com.liferay.commerce.model.CommerceOrder;
+import com.liferay.commerce.order.status.CommerceOrderStatus;
+import com.liferay.commerce.order.status.CommerceOrderStatusRegistry;
+import com.liferay.commerce.product.model.CommerceChannel;
+import com.liferay.commerce.product.service.CommerceChannelLocalService;
+import com.liferay.commerce.service.CommerceOrderLocalService;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.settings.GroupServiceSettingsLocator;
+
+import java.util.Locale;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+
+/**
+ * @author Brian I. Kim
+ */
+@Component(
+	property = {
+		"commerce.order.status.key=" + QuoteProcessedCommerceOrderStatusImpl.KEY,
+		"commerce.order.status.priority:Integer=" + QuoteProcessedCommerceOrderStatusImpl.PRIORITY
+	},
+	service = CommerceOrderStatus.class
+)
+public class QuoteProcessedCommerceOrderStatusImpl
+	implements CommerceOrderStatus {
+
+	public static final int KEY =
+		CommerceOrderConstants.ORDER_STATUS_QUOTE_PROCESSED;
+
+	public static final int PRIORITY = 50;
+
+	@Override
+	public CommerceOrder doTransition(CommerceOrder commerceOrder, long userId)
+		throws PortalException {
+
+		commerceOrder.setOrderStatus(KEY);
+
+		commerceOrder = _commerceOrderLocalService.updateCommerceOrder(
+			commerceOrder);
+
+		return _commerceOrderLocalService.updateCommerceOrder(commerceOrder);
+	}
+
+	@Override
+	public int getKey() {
+		return KEY;
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(
+			locale, CommerceOrderConstants.getOrderStatusLabel(KEY));
+	}
+
+	@Override
+	public int getPriority() {
+		return PRIORITY;
+	}
+
+	@Override
+	public boolean isEnabled(CommerceOrder commerceOrder)
+		throws PortalException {
+
+		CommerceOrderFieldsConfiguration commerceOrderFieldsConfiguration =
+			_getCommerceOrderFieldsConfiguration(commerceOrder);
+
+		if (commerceOrderFieldsConfiguration.requestQuoteEnabled()) {
```

Comment:
We will have products that require a quote regardless of this setting, so I don't think this check is ok

Reply:
This is also true for the RequestQuote status